staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission_test.go

```diff
@@ -277,11 +277,19 @@ func TestAdmit(t *testing.T) {
 			// ErrWebhookRejected is not an error for our purposes
 			if tt.errorContains != "" {
 				if err == nil || !strings.Contains(err.Error(), tt.errorContains) {
-					t.Errorf(" expected an error saying %q, but got %v", tt.errorContains, err)
+					t.Errorf("expected an error saying %q, but got %v", tt.errorContains, err)
+				}
+			}
+			for _, externalAdmissionHook := range tt.hookSource.hooks {
+				if externalAdmissionHook.Name != "nomatch" {
```

Comment:
This is fragile, someone might change the first test case name. Also I can imagine adding test cases where a different client config is used. How about writing a separate test for the client cache?

Reply:
ok